lib/ModuleManager.cpp

```diff
@@ -836,6 +848,21 @@ void QoreModuleManager::loadModuleIntern(ExceptionSink& xsink, const char* name,
             }
         }
 
+        // check whether it is a module folder
+        QoreString modulePath(*w);
+        modulePath += QORE_DIR_SEP_STR;
+        modulePath += name;
+
+        if (QoreDir::folder_exists(modulePath, xsink)) {
+            mi = loadSeparatedModule(xsink, modulePath, name, pgm, reexport, pholder.release(), load_opt & QMLO_REINJECT ? mpgm : 0, load_opt);
+
+            if (xsink)
+                assert(!mi);
```

Comment:
assert is removed in debug build, so first of all, I would add `{` around it, and second, what is the point of asserting if there was an exception? it will still call the `qore_checl_load_module_intern` right?

Reply:
resolved by moving xsink and asserts checks to qore_checl_load_module_intern